### DISCUSSION TITLE
Clock() and tick() not working with a setTimeout

### DISCUSSION THREAD

**Discussion by rebeccarothschild** [cypress-io/cypress#24196] [Questions and Help]
Hi there, 

I'm E2E testing some application code that has a setTimeout call that delays a network request which then triggers a component to display. I'm trying to speed up time to skip the timeout but I haven't gotten it to work. Is there some limitations to how clock and tick can work with async methods? Below is an example of what I'm trying to do.

spec.js:

```
describe(
  "Delay test",
  function() {
    it("Verify component visible", function() {
      cy.clock();
      cy.visit("/");
      cy.login(userName, password);
      cy.waitToCompleteLoading();
      cy.tick(600000);
      cy.verifyComponentPresent();
    });
  }
);
```

setTimeout call: 
```
fetchObject() {
        const component = this;

        setTimeout(() => {
          // Delay fetch
          component.$store.dispatch(`/${GET_OBJECT}`); // async function
        }, 600000);
    }
```

Using **Cypress 10.3.0**

**Comment by RiZKiT** [comment]
Looks good so far, I just implemented something like this because I found this problem here and it works for me, fast forward a `setTimeOut()` .

I have two suggestion:
- try `cy.tick(600000 * 2)`, to make sure the timeout is definitely triggered
- if `cy.login()` is implemented with a new pageload, move `cy.clock()` between `cy.visit()` and `cy.login()`
